port/zos390/omrsl.c

```diff
@@ -167,13 +255,38 @@ omrsl_close_shared_library(struct OMRPortLibrary *portLibrary, uintptr_t descrip
 uintptr_t
 omrsl_lookup_name(struct OMRPortLibrary *portLibrary, uintptr_t descriptor, char *name, uintptr_t *func, const char *argSignature)
 {
-	void *address;
-	dllhandle *handle;
+	void *address = NULL;
+	dllhandle *handle = NULL;
 
 	Trc_PRT_sl_lookup_name_Entry(descriptor, name, argSignature);
 
-	handle = (dllhandle *)descriptor;
-	address = (void *)dllqueryfn(handle, name);
+#if defined(J9ZOS39064)
+	if (OMR_ARE_ALL_BITS_SET(descriptor, OMRPORT_SL_ZOS_31BIT_TARGET_HIGHTAG)) {
+		OMR_CEL4RO31_infoBlock *ro31InfoBlock = NULL;
+
+		/* 31-bit target library */
+		handle = (dllhandle *)(OMRPORT_SL_ZOS_31BIT_TARGET_MASK & descriptor);
+		DMESSAGE(("omrsl_lookup_name: Attempt to query 31-bit DLL function [%s] from DLL handle: %p\n", name, handle))
+		ro31InfoBlock = omr_cel4ro31_init(OMR_CEL4RO31_FLAG_QUERY_TARGET_FUNC, NULL, name, 0);
+		if (NULL != ro31InfoBlock) {
+			OMR_CEL4RO31_controlBlock *ro31ControlBlock = &(ro31InfoBlock->ro31ControlBlock);
+			ro31ControlBlock->dllHandle = (uint32_t)handle;
+			omr_cel4ro31_call(ro31InfoBlock);
+			if (OMR_CEL4RO31_RETCODE_OK == ro31ControlBlock->retcode) {
+				/* CEL4RO31 returns the function descriptor (FDCB) in the functionEntryPoint member */
+				address = (void *)ro31ControlBlock->functionEntryPoint;
+				address = (void* )(OMRPORT_SL_ZOS_31BIT_TARGET_HIGHTAG | (uintptr_t)address);
```

Comment:
```suggestion
				address = (void *)(OMRPORT_SL_ZOS_31BIT_TARGET_HIGHTAG | (uintptr_t)address);
```

Reply:
Thanks for catching this.  Fixed under 813bcfe.